tests/test_dynamic_publisher_block.py

```diff
@@ -152,3 +152,23 @@ def test_closing(self):
             event.wait(.3)
 
             pub.return_value.close.assert_called_once()
```

Comment:
The `assert_called_once` assertion is a newer assertion (since Python 3.6), that's why the CI tests are failing. They run in 3.4 or 3.5.

Use one of these instead:
```python
pub.return_value.close.assert_called_once_with()
```
or
```python
self.assertEqual(pub.return_value.close.call_count, 1)
```